### ISSUE TITLE
TypeError in development_runner.py

### ISSUE THREAD

**Issue by cmin764** [timothycrosley/hug#694]
With 2.4.1 on 3.7.0, you can reproduce this by just running `hug -f t.py` over the following script:

```python
import hug

@hug.get("/test")
def test_func(r: dict):
    print(r)
    return {"r": r}
```

and gives you this (I guess it behaves the same with any kind of run, even the `--module` one):

```bash
$ hug -f t.py
Unhandled exception in thread started by <function reload_checker at 0x10324e0d0>
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/hug/development_runner.py", line 108, in reload_checker
    if path[-4:] in ('.pyo', '.pyc'):
TypeError: 'NoneType' object is not subscriptable
```

L.E.: the problem doesn't seem to appear if you also supply the `--manual_reload True` parameter.

**Comment by cmin764** [comment]
I think it's linked to this PR: https://github.com/timothycrosley/hug/pull/691/files

**Comment by RobertWan91** [comment]
This error also occurs to me. However, it doesn't affect the running of the code. I can still use Insomnia to test the HUG API.

**Comment by bargitta** [comment]
Meet the same issue, any idea how to fix it?

**Comment by Dextroz** [comment]
Issue still persists. Python 3.7.2, Hug 2.4.1.

**Comment by GhostofGoes** [comment]
This should be fixed in `develop`, but the change hasn't yet been published to PyPI.

**Comment by voglster** [comment]
For those of us who got here because they were seeing this error and hot reloading stopped working..
Click the pr and apply that patch (2 lines) until it gets published to PyPi
Specifically add
```
if not path:
  continue
```
 to line 108 directly after  `path = getattr(module, '__file__', '')`

reloading should work again until its released on PyPI

**Comment by GhostofGoes** [comment]
`pip install https://github.com/timothycrosley/hug/archive/develop.tar.gz`

**Comment by GhostofGoes** [comment]
This is fixed in 2.4.2
